### PR TITLE
openSSL compatibility API for EVP, BIO, and SSL_SESSION

### PR DESCRIPTION
**Pull request by embhorn**
Adding support for the following:

- BIO_f_md
- BIO_get_md_ctx
- BIO_set_close
- BIO_should_retry
- BIO_vprintf
- ERR_print_errors_cb
- EVP_aes_128_cfb1
- EVP_aes_192_cfb1
- EVP_aes_256_cfb1
- EVP_aes_128_cfb8
- EVP_aes_256_cfb8
- EVP_aes_192_cfb8
- EVP_aes_128_cfb128
- EVP_aes_192_cfb128
- EVP_aes_256_cfb128
- EVP_aes_128_ofb
- EVP_aes_192_ofb
- EVP_aes_256_ofb
- EVP_aes_128_xts
- EVP_aes_256_xts
- EVP_CIPHER_CTX_reset
- EVP_mdc2
- EVP_MD_CTX_reset
- OpenSSL_add_ssl_algorithms
- SSL_MODE_AUTO_RETRY
- SSL_SESSION_get0_peer

### CODE REVIEW

File: wolfcrypt/src/aes.c [hunk 1]
```diff
@@ -7072,43 +7079,66 @@ int wc_AesEcbDecrypt(Aes* aes, byte* out, const byte* in, word32 sz)
...
-/* CFB 128
+#if defined(WOLFSSL_AES_CFB) || defined(WOLFSSL_AES_OFB)
+/* Feedback AES mode
  *
  * aes structure holding key to use for encryption
  * out buffer to hold result of encryption (must be at least as large as input
  *     buffer)
  * in  buffer to encrypt
  * sz  size of input buffer
+ * pre flag to xor after or before feedback. If 1 then add feedback before xor
```
Comment (JacobBarthelmeh) [inline wolfcrypt/src/aes.c]: I missed changing this to "mode" pre was an earlier parameter name

File: wolfssl/ssl.h [hunk 2]
```diff
@@ -1978,6 +1988,11 @@ WOLFSSL_API int wolfSSL_want(WOLFSSL*);
 WOLFSSL_API int wolfSSL_want_read(WOLFSSL*);
 WOLFSSL_API int wolfSSL_want_write(WOLFSSL*);
 
+#ifndef NO_FILESYSTEM
+#include <stdarg.h> /* var_arg */
```
[the remark this answers is not in the document]
Comment (embhorn) [reply wolfssl/ssl.h]: Check

File: wolfcrypt/src/aes.c (hunk 1 above)
Comment (embhorn) [reply wolfcrypt/src/aes.c]: Check

**Comment by embhorn** [comment]
Hi @JacobBarthelmeh ,

Please have a look at the changes since https://github.com/wolfSSL/wolfssl/pull/2775/commits/a64e1540ba0f03b5eb16d441372f6f04b885e456
I added EVP support for SHA3, as well as updating for the recent EVP overhaul.

**Pull request merged**